tests/ha/control_pod/test_control_pod_restart.py

```diff
@@ -747,161 +747,97 @@ def test_mpu_after_ctrl_pod_failover(self):
-                        failover_node, cnt)
-            resp = self.ha_obj.change_pod_node(self.node_master_list[0],
-                                               pod_node={control_pod_name: failover_node})
-            assert_utils.assert_true(resp[0], resp)
-            LOGGER.info("Check cluster status")
+        for count in range(num_users):
+            resp = self.ha_obj.ha_s3_workload_operation(s3userinfo=list(users.values())[count],
+                                                        log_prefix=self.test_prefix,
+                                                        skipcleanup=True)
+            assert_utils.assert_true(resp[0], resp[1])
+        LOGGER.info("Step 1: Performed WRITEs-READs-Verify with variable sizes objects for %s IAM "
+                    "users created.", num_users)
+        LOGGER.info("Step 2: Shutdown %s control pods in loop", self.repl_num - 1)
+        LOGGER.info("Get the list of control pods")
+        pod_list = self.node_master_list[0].get_all_pods(pod_prefix=const.CONTROL_POD_NAME_PREFIX)
+        pod_left = None
+        for loop in range(len(pod_list)):
+            LOGGER.info("Shutting down %s control pods for loop: %s", self.repl_num - 1, loop)
+            delete_pods = list()
+            while pod_left not in delete_pods:
+                delete_pods.extend(self.system_random.sample(pod_list, self.repl_num - 1))
+            for pod in delete_pods:
+                resp = self.node_master_list[0].create_pod_replicas(num_replica=0, pod_name=pod)
+                assert_utils.assert_true(resp[0], resp[1])
+            LOGGER.info("Step 3: Check cluster status")
             resp = self.ha_obj.poll_cluster_status(self.node_master_list[0])
-            assert_utils.assert_true(resp[0], resp)
-            LOGGER.info("Step 3: Successfully failed over control pod to %s. Cluster is in good "
-                        "state", failover_node)
-
-            self.restore_pod = self.deploy = True
-            LOGGER.info("Step 4: Verify if IAM users %s are persistent across control pod failover",
-                        uids)
-            for user in uids:
+            assert_utils.assert_true(resp[0], resp[1])
+            LOGGER.info("Step 3: Cluster status is online")
+            LOGGER.info("Step 4: Create multiple IAM users and delete randomly selected IAM users")
+            users_loop = self.mgnt_ops.create_account_users(nusers=num_users)
+            self.s3_clean = users_loop
+            created_list = list(users_loop.keys())
+            num = self.system_random.randint(1, num_users)
+            delete_list = self.system_random.sample(created_list, num)
+            for buck in range(num):
+                resp = self.ha_obj.delete_s3_acc_buckets_objects(delete_list[buck])
+                assert_utils.assert_true(resp[0], resp[1])
+                created_list.remove(delete_list[buck])
+            for user in created_list:
```

Comment:
created_list needs to be kept global for this test.
As number of users which we are not deleting at line 799, should be persistent across this outer for loop.
So we will have to extend created_list in each loop, and remove whichever users are deleted, from the list.